### PR TITLE
Final (?) pane-renaming fix

### LINKED ISSUE #6770: [BUG]: After adding a file in tree view, the added file isn't visible in shared links
### Describe the bug

-

### Steps to reproduce

1. Open 'Templates'/'C++ Cmake'
2. From the tree pane menu 'Add'/'Source editor'
A file named `C++ source #5` appears both in the editors and in the tree (under 'excluded files').
3. Share this CE instance, either as full or short url,
4. Open the shared url in a new tab.

`C++ source #5` is still visible in the editors, but not in the tree view.  If, however, you 'Add'/'Source editor' again - it reappears, alongside the newly added source.

### Expected behavior

Should be visible in tree whenever it exists.

### Reproduction link

https://www.compiler-explorer.com/z/Kaq4acrq9

### Screenshots

image

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by OfekShilon**
Some major-ish fixes to tree and minor fixes in multifile-service.
This hopefully fixes #6770, #5734, and maybe others.

### CODE REVIEW

**Comment by OfekShilon** [comment]
This will be applied *after* the previous 3 PRs.

File: static/panes/tree.ts [hunk 1]
```diff
@@ -445,6 +452,13 @@ export class Tree {
 
         dragSource.on('click', () => {
             this.hub.addInEditorStackIfPossible(dragConfig.bind(this));
+            // at this point the editor is initialized with default contents
```
Comment (partouf) [inline static/panes/tree.ts:455]: This added code doesn't make much sense to me. This is not the only place where you can add editors and at this point there is no actual physical editor yet which you can query. What does this solve what the onEditorOpen cannot catch?

Comment (OfekShilon) [reply static/panes/tree.ts:455]: `onEditorOpen` was indeed the first choice for this code, but during its execution the `file` retrieved by `multifileService.getFileByEditorId` has no `filename` and no `content`. These are set by the `Editor` ctor, a bit later.

Comment (OfekShilon) [reply static/panes/tree.ts:455]: @partouf  LGTU?

Comment (OfekShilon) [reply static/panes/tree.ts:455]: I will go ahead and merge, can fix later whatever comes up.

**Pull request merged**